modules/payment/src/Entity/PaymentGateway.php

```diff
@@ -159,4 +161,67 @@ protected function getPluginCollection() {
     return $this->pluginCollection;
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public function postSave(EntityStorageInterface $storage, $update = TRUE) {
```

Comment:
I feel like the order in the code should be: 
preSave
postSave
preDelete
postDelete

No?